src/components/MultipleCheckboxOptions.tsx

```diff
@@ -0,0 +1,55 @@
+import { Box, Checkbox, VStack } from '@chakra-ui/react';
+import { useState, type FunctionComponent } from 'react';
+
+export type MultipleCheckboxOptionsProps = {
+  options: string[];
+  onChange: (selectedOptions: string[]) => void;
+};
+
+/**
+ * A box with multiple checkbox options.
+ *
+ * @param props - The component props.
+ * @param props.options - The options to be shown with checkboxes.
+ * @param props.onChange - A function to be called when option selection is changed.
+ * @returns A React component.
+ */
+export const MultipleCheckboxOptions: FunctionComponent<
+  MultipleCheckboxOptionsProps
+> = ({ options, onChange, ...props }) => {
+  const [selectedOptions, setSelectedOptions] = useState<string[]>([]);
+
+  const handleCheckboxChange = (option: string) => {
+    const updatedSelection = selectedOptions.includes(option)
+      ? selectedOptions.filter((item) => item !== option)
+      : [...selectedOptions, option];
+
+    setSelectedOptions(updatedSelection);
+    onChange(updatedSelection);
+  };
```

Comment:
Not a must, but suggest

```suggestion
  const [checkedItems, setCheckedItems] = React.useState(
    options.map((e) => false)
  );
  const handleCheckboxChange = (checked: boolean, index:number) => {
    checkedItems[index] = checked;
    setCheckedItems([...checkedItems]);
    onChange(checkedItems);
  };
```

in that way, you can simplify the checkbox, with out a n square loop to search if a checkbox checked

```
<Checkbox
            size="md"
            borderRadius="0.25rem"
            padding="0.012rem"
            marginInline="1"
            key={option}
            isChecked={checkedItems[idx]}
            onChange={(e) => handleCheckboxChange(e.target.checked, idx)}
          >
```

Reply:
fixed